assets/src/blocks/Gallery/GalleryCarousel.js

```diff
@@ -1,16 +1,33 @@
-import { useState, useEffect, useRef } from '@wordpress/element';
+import { useState, useEffect, useRef, memo } from '@wordpress/element';
 import _uniqueId from 'lodash/uniqueId';
+import { Lightbox, wpImagesToItems } from '../../components/Lightbox/Lightbox';
 
 const { __ } = wp.i18n;
 
 // This will trigger the browser to synchronously calculate the style and layout
 // You can find a list of examples here: https://gist.github.com/paulirish/5d52fb081b3570c81e3a
 const reflow = element => element.offsetHeight;
 
+const MemoLightbox = memo(Lightbox, (prevProps, nextProps) => {
+  return prevProps.id === nextProps.id && prevProps.isOpen === nextProps.isOpen;
+});
+
 export const GalleryCarousel = ({ images }) => {
   const [currentSlide, setCurrentSlide] = useState(0);
   const [sliding, setSliding] = useState(false);
+  const [lightboxIsOpen, setLightboxIsOpen] = useState(false);
+  const [lightboxIndex, setLightboxIndex] = useState(0);
+
+  const onImageClick = lightboxIndex => {
+    setLightboxIsOpen(true);
+    setLightboxIndex(lightboxIndex);
+  };
+
+  const onLightboxClose = () => {
+    setLightboxIsOpen(false);
+  };
 
+  const items = wpImagesToItems(images);  
```

Comment:
I guess we could assume that these images will be the same structure in all places? Or if not we could make them so. As they basically come from the same data structure, in theory they should all have the same fields.

Then we could handle this mapping inside of the lightbox in one place, and can just pass the images to it instead.

Reply:
Not sure, the caption for the images can come from different places. In the case of image blocks, it comes from the DOM. I'll take a look.

Reply:
I moved this function to `GalleryFrontend`, so it's  in a single place for galleries, but as said in the previous comment, building the items for image blocks is different, and it could be different for other use cases of the Lightbox, so I'd rather make the Lightbox expect the `items` collections it needs without doing any transforms internally.